server/src/main/java/org/opensearch/indices/replication/copy/SegmentReplicationPrimaryService.java

```diff
@@ -165,7 +167,12 @@ public void messageReceived(StartReplicationRequest request, TransportChannel ch
             logger.trace("Received request for checkpoint {}", checkpoint);
             final CopyState copyState = getCopyState(checkpoint);
             channel.sendResponse(
-                new TransportCheckpointInfoResponse(copyState.getCheckpoint(), copyState.getMetadataSnapshot(), copyState.getInfosBytes())
+                new TransportCheckpointInfoResponse(
+                    copyState.getCheckpoint(),
+                    copyState.getMetadataSnapshot(),
+                    copyState.getInfosBytes(),
+                    copyState.getPendingDeleteFiles()
```

Comment:
Nitpick - add a comment on why we're using the list of files pending delete as the additional files list

Reply:
updated TransportCheckpointInfoResponse to not generic & added detail there.